src/goto-instrument/contracts/assigns.cpp

```diff
@@ -99,7 +99,8 @@ assigns_clauset::assigns_clauset(
 
 void assigns_clauset::add_target(const exprt &target_expr)
 {
-  auto insertion_succeeded = targets.emplace(*this, target_expr).second;
+  auto insertion_succeeded =
+    write_set.emplace(*this, targett::normalize(target_expr)).second;
```

Comment:
I am not sure why this change is needed.
The constructor already calls `normalize` before storing the `address`.

I think this call should remain as before and we should be able to remove `expr` from `targett` entirely -- only `address` seems to be enough to track everything.

Reply:
Correct, we don't need to normalize this here again and we can completely remove `expr`. Done.